staticaddr/manager.go

```diff
@@ -0,0 +1,152 @@
+package staticaddr
+
+import (
+	"context"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop"
+	"github.com/lightninglabs/loop/staticaddr/script"
+	staticaddressrpc "github.com/lightninglabs/loop/swapserverrpc"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+type ManagerConfig struct {
+	AddressClient staticaddressrpc.StaticAddressServerClient
+
+	LoopClient *loop.Client
+
+	Store AddressStore
+
+	Lnd lndclient.LndServices
+}
+
+// AddressManager manages the address state machines.
+type AddressManager struct {
+	cfg *ManagerConfig
+
+	runCtx context.Context
+
+	initChan chan struct{}
+
+	sync.Mutex
+}
+
+// NewAddressManager creates a new address manager.
+func NewAddressManager(cfg *ManagerConfig) *AddressManager {
+	return &AddressManager{
+		cfg:      cfg,
+		initChan: make(chan struct{}),
+	}
+}
+
+// Run runs the address manager.
+func (m *AddressManager) Run(ctx context.Context) error {
```

Comment:
Will the `AddressManager` have a runloop? If not, then a simple `Init` call would suffice (and also no need for storing the `runCtx`).

Reply:
Yes, the manager will have a run loop to handle different actions of a static address like timeout, confirmations, coop-spends and loopins.

Reply:
If that's coming in follow up PR's then perhaps worth thinking about whether we need the embedded context there.

Reply:
I've removed the embedded context for now.